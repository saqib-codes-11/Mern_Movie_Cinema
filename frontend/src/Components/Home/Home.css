
.imgContainer {
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 80%;
}

.imaxContainer {
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 150%;
    height: 200%;
    padding-right: 150px;
}

#home {
    display: table;
    position: relative;
    width: 100%;
    background-size: cover;
    padding-bottom: 25px;
}

.overlay{
    color:#fff;
  position:absolute;
  z-index:12;
  top:40%;
  left:0;
  width:100%;
  text-align:center;
}

.carousel-item:after {
  content:"";
  display:block;
  position:absolute;
  top:0;
  bottom:0;
  left:0;
  right:0;
  background:rgba(0,0,0,0.4);
}