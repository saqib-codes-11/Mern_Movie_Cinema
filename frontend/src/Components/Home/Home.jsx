
import { Carousel, Container, Col, Row } from 'react-bootstrap';
import InTheHeights from '../../Images/In-The-Heights.jpg';
import BlackWidow from '../../Images/Black-Widow.jpg';
import F9 from '../../Images/Fast9.jpg';
import DemonSlayer from '../../Images/Demon Slayer.jpg';
import IMAX from '../../Images/Imax.png';
import IMAXScreen from '../../Images/Imax screen.jpg';
import '../../CSS/Pages.css';
import './Home.css';
const Home = () => {
    return (
        <div class="background">
            <div class="bgBlur">
                <div class="overlay">
                    <h1 class='landing-text'> Welcome to QA Cinemas</h1>
                    <h3 class="landing-text">Experience all the latest movies in stunning IMAX!</h3>
                </div>
                <Carousel className="d-flex align-items-center justify-content-center">
                    <Carousel.Item interval={4000}>
                        <img
                            style={{ width: "100%", objectFit: "cover" }}
                            src={InTheHeights}
                            alt="In the heights img"
                            href="/Forum"
                            height="900"
                        />
                    </Carousel.Item>
                    <Carousel.Item interval={4000}>
                        <img
                            style={{ width: "100%", objectFit: "cover" }}
                            src={BlackWidow}
                            alt="Black Widow img"
                            height="900"
                        />
                    </Carousel.Item>
                    <Carousel.Item interval={4000}>
                        <img
                            style={{ width: "100%", objectFit: "cover" }}
                            src={DemonSlayer}
                            alt="Demon Slayer img"
                            height="900"
                        />
                    </Carousel.Item>
                    <Carousel.Item interval={4000}>
                        <img
                            style={{ width: "100%", objectFit: "cover" }}
                            src={F9}
                            alt="Fast and Furious img"
                            height="900"
                        />
                    </Carousel.Item>
                </Carousel>
                <Container>
                    <Container>
                        <Row className="mt-5">
                            <Col md={6}>
                                <h2 class="display-4">Unrivaled Quality</h2>
                                <p class="lead">
                                    Immerse yourself completely in the big screen experience with QA Cinemas IMAX. Every aspect of our IMAX Screens is designed to pull you into the film. Indulge in the journey that tickles your senses and brings them back to life. Transporting yourself into a new reality with our super-sized screens pulling you into the film, ear-tingling audio environment to ensure you can even hear a pin drop and where within the film as well as breath-taking imagery, watching movies on your computer will never be the same again.
                                </p>
                            </Col>
                            <Col md={6} className="mt-5">
                                <img src={IMAXScreen} alt="Imax Screen img" class="imgContainer" />
                            </Col>
                            <Col md={12}>
                                <div>
                                    <img src={IMAX} alt="Imax Logo" class="imgContainer" />
                                </div>
                            </Col>
                        </Row>
                    </Container>
                </Container>
            </div>
        </div>


    )
}
export default Home;