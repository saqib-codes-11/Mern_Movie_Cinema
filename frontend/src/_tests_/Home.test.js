import React from 'react';
import Home from '../Components/Home/Home';
import renderer from 'react-test-renderer';

test("It matches the snapshot", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();

})