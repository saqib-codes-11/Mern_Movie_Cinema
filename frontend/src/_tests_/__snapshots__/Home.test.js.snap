// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the snapshot 1`] = `
<div
  class="background"
>
  <div
    class="bgBlur"
  >
    <div
      class="overlay"
    >
      <h1
        class="landing-text"
      >
         Welcome to QA Cinemas
      </h1>
      <h3
        class="landing-text"
      >
        Experience all the latest movies in stunning IMAX!
      </h3>
    </div>
    <div
      className="d-flex align-items-center justify-content-center carousel slide"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <ol
        className="carousel-indicators"
      >
        <li
          className="active"
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
      </ol>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
          interval={4000}
        >
          <img
            alt="In the heights img"
            height="900"
            href="/Forum"
            src="In-The-Heights.jpg"
            style={
              Object {
                "objectFit": "cover",
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="carousel-item"
          interval={4000}
        >
          <img
            alt="Black Widow img"
            height="900"
            src="Black-Widow.jpg"
            style={
              Object {
                "objectFit": "cover",
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="carousel-item"
          interval={4000}
        >
          <img
            alt="Demon Slayer img"
            height="900"
            src="Demon Slayer.jpg"
            style={
              Object {
                "objectFit": "cover",
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="carousel-item"
          interval={4000}
        >
          <img
            alt="Fast and Furious img"
            height="900"
            src="Fast9.jpg"
            style={
              Object {
                "objectFit": "cover",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <a
        className="carousel-control-prev"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
      <a
        className="carousel-control-next"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </div>
    <div
      className="container"
    >
      <div
        className="container"
      >
        <div
          className="mt-5 row"
        >
          <div
            className="col-md-6"
          >
            <h2
              class="display-4"
            >
              Unrivaled Quality
            </h2>
            <p
              class="lead"
            >
              Immerse yourself completely in the big screen experience with QA Cinemas IMAX. Every aspect of our IMAX Screens is designed to pull you into the film. Indulge in the journey that tickles your senses and brings them back to life. Transporting yourself into a new reality with our super-sized screens pulling you into the film, ear-tingling audio environment to ensure you can even hear a pin drop and where within the film as well as breath-taking imagery, watching movies on your computer will never be the same again.
            </p>
          </div>
          <div
            className="mt-5 col-md-6"
          >
            <img
              alt="Imax Screen img"
              class="imgContainer"
              src="Imax screen.jpg"
            />
          </div>
          <div
            className="col-md-12"
          >
            <div>
              <img
                alt="Imax Logo"
                class="imgContainer"
                src="Imax.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
